tests/lvmo/test_lvm_snapshot_node_reboot.py

```diff
@@ -0,0 +1,308 @@
+        logger.info(f"Rebooting {nodes_names}")
+        nodes.restart_nodes(nodes_ocs, force=False, wait=False)
+        logger.info(f"Waiting for nodes {nodes_names} to be in READY state")
+        wait_for_nodes_status(node_names=nodes_names, timeout=300)
+        logger.info(f"Nodes {nodes_names} in ready state")
+        lvm.wait_for_lvm_pod_running()
+
+        @retry(CommandFailed, tries=30, delay=3)
+        def switch_project_after_reboot():
+            switch_to_project(project_name=proj_obj.namespace)
+
+        switch_project_after_reboot()
+
+        if not wait_for_pods_to_be_running(
+            pod_names=pod_name_list, sleep=3, namespace=proj_obj.namespace
+        ):
+            raise PodDidNotReachRunningState
+
+        lvm.compare_percent_data_from_pvc(pvc_restore, fio_size)
+
+        restored_pod_obj.run_io(
+            storage_type,
+            size=f"{second_fio_size}g",
+            rate="1500M",
+            runtime=0,
+            rate_process=None,
+            fio_filename=f"second-{filename}",
+            buffer_pattern="0xdeadface",
+            direct=1,
+            invalidate=0,
+            bs="100M",
+            jobs=1,
+            readwrite=readwrite,
+        )
+
+        restored_pod_obj.get_fio_results()
+
+        if block:
+            if fio_size > second_fio_size:
+                after_expected_util = (
+                    float(fio_size + second_fio_size) / float(thin_pool_size) * 100
+                )
+
+                lv_util_after_second_fio = float(fio_size)
+            else:
+                after_expected_util = (
+                    float(second_fio_size + fio_size) / float(thin_pool_size) * 100
+                )
+
+                lv_util_after_second_fio = float(second_fio_size)
+
+        else:
+            after_expected_util = (
+                float(fio_size + second_fio_size) / float(thin_pool_size)
+            ) * 100
+            lv_util_after_second_fio = float(fio_size + second_fio_size)
+
+        lvm.compare_percent_data_from_pvc(pvc_restore, lv_util_after_second_fio)
+        lvm.compare_thin_pool_data_percent(
+            data_percent=after_expected_util,
+            sampler=True,
+            timeout=10,
+            wait=1,
+        )
+        pvc_restore_second_start_time = time.time()
+        pvc_restore_second = snapshot_restore_factory(
+            snapshot,
+            storageclass=sc_obj.name,
+            restore_pvc_name=f"{pvc_obj.name}-second-restore",
+            size=str(pvc_size * 1024 * 1024 * 1024),
+            volume_mode=volume_mode,
+            restore_pvc_yaml=constants.CSI_LVM_PVC_RESTORE_YAML,
+            access_mode=access_mode,
+            status=status,
+            timeout=300,
+        )
+        pvc_restore_second_end_time = time.time()
+        logger.info(
+            f"Pvc creation time is {(pvc_restore_second_end_time - pvc_restore_second_start_time)}"
+        )
+        pod_3_start_time = time.time()
```

Comment:
Instead of pod1, pod2, pod3, why can't we iterate over 3 pods with a for loop?

Reply:
I didn't called the pod 1,2,3. Only the timing which is nice to know the time after reboot, before reboot and so on.
The pods are created in different stages of the test.
1. origin pod before reboot
2. pod with restored snapshot before reboot IO and after reboot more IO.
3. pod with restored snapshot created after reboot. 
So the loop is not possible.
In the logging I'm using pods names
`f"Pod {restored_pod_obj_second.name} got to running state after {pod_3_elapsed_time}"`

Reply:
Ok